### DISCUSSION TITLE
Usage with Timber / Twig template engine?

### DISCUSSION THREAD

**Discussion by jasalt** [svandragt/htmxpress#16] [Ideas]
I recently found out about Timber which is a layout or theme framework for WordPress using common patterns from server side rendering frameworks https://timber.github.io/docs/v2/. While I'm still getting up to speed with it rebuilding some slow sites, it seems like it could be a nice fit to use together with HTMX.

@svandragt would you have interest and/or time to check that out a bit and maybe give some pointers how it would integrate with HTMX(Press)? I may look into it myself a bit later also, but I don't have as good understanding on the WP internals being used.

This is also somewhat related to issue https://github.com/svandragt/htmxpress/issues/1 as Twig templating it uses is very similar to eg. Django or Jinja2 templates, maybe many patterns from there could be used in WP also with more similar template engine on both sides.

EDIT: there was recent issue about this in Timber repo also: https://github.com/timber/timber/issues/2944

**Comment by yurimatheus** [comment]
Hello colleagues, sorry for the intrusion in the post, but I was very happy
to see Timber and the link from our colleague Jarkko.

It uses an MVC pattern, something we don't even use when creating templates
or plugins. In addition to its lean structure.

I wish I could help with something.

Em dom., 26 de mai. de 2024 02:37, Jarkko Saltiola ***@***.***>
escreveu:

**Reply by jasalt** [reply]
@yurimatheus I have had some initial success using another plugin similar to Htmxpress in conjunction with Timber Starter Theme which I documented some in a fork here https://github.com/jasalt/starter-theme/blob/2.x-htmx-test/README.md.
